### PR TITLE
cmd/juju/storage: add "list", tweak "show"

### PR DESCRIPTION
**Pull request by axw**
This commit introduces "juju storage list" to list
all of the storage instances in the environment.
The default output is a tabular format, similar to
"juju status --format tabular".

The existing "juju storage show" has been tweaked
to output as a map, rather than a slice. This makes
the output more like "juju status". We also translate
tags to IDs (unit/service name, storage ID), as tags
are not intended for users' eyes.

The StorageName value is now computed from the storage
ID. Finally, the "location", "available-size" and
"total-size" fields are omitted if unavailable. To
effect this, we make their types pointers, so they
are nullable.

This commit requires https://github.com/juju/names/pull/38

### CODE REVIEW

File: cmd/juju/storage/list.go [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+)
+
+const ListCommandDoc = `
+List information about storage instances.
+
+options:
+-e, --environment (= "")
+   juju environment to operate in
+-o, --output (= "")
+   specify an output
```
Comment (wallyworld) [inline cmd/juju/storage/list.go]: an output file?

File: cmd/juju/storage/list.go [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+// Init implements Command.Init.
+func (c *ListCommand) Init(args []string) (err error) {
+	return cmd.CheckEmpty(args)
+}
+
+// Info implements Command.Info.
+func (c *ListCommand) Info() *cmd.Info {
+	return &cmd.Info{
+		Name:    "list",
+		Purpose: "lists storage instance",
```
Comment (wallyworld) [inline cmd/juju/storage/list.go]: instances?

**Comment by wallyworld** [comment]
LGTM, very nice

File: cmd/juju/storage/list.go (hunk 1 above)
Comment (axw) [inline cmd/juju/storage/list.go]: Indeed. Fixed here and in show.go, where I copied from :)

File: cmd/juju/storage/list.go (hunk 2 above)
Comment (axw) [inline cmd/juju/storage/list.go]: Done.

**Pull request merged**